backend/logic/courses/courses.js

```diff
@@ -261,12 +261,104 @@ var updateCourse = async (course, newTitle, newTags) => {
   }
 };
 
+/**
+ * Assign a course to a curriculum
+ * @author Yufei Liu
+ * @param {string} curriculumType,{string} courseCode,{string} curriculum
+ * @returns true if successful
+ * @throws error if curriculumType, courseCode or curriculum dose not exist
+ *         course already assigned a curriculum
+ *
+ */
+var assignCourseToCurriculum = async (
+  curriculumType,
+  courseCode,
+  curriculum
+) => {
+  let conn = await mysql.getNewConnection();
+
+  const checkExistQuery = `SELECT *
+  FROM (
+      SELECT course_code, curriculum_name FROM curriculum_tech_comps
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_complementaries
+      union all
+      SELECT course_code, curriculum_name FROM curriculum_core_classes
+  ) a
+  WHERE course_code = ? AND curriculum_name = ?;`;
+
+  // Check if the course has already been assigned, verify if curriculum type, course and curriculum are existed in DB
+  let ifAssigned;
+  let checkCourse;
+  let checkCurriculum;
+
+  try {
+    ifAssigned = await conn.query(checkExistQuery, [courseCode, curriculum]);
+    checkCourse = await conn.query(
+      "SELECT * FROM courses WHERE course_code = ?",
+      [courseCode]
+    );
+    checkCurriculum = await conn.query(
+      "SELECT * FROM curriculums WHERE curriculum_name = ?",
+      [curriculum]
+    );
+  } catch (err) {
+    console.log(err);
+    conn.release();
+    throw Error("Internal Server Error!\n");
+  }
+
+  if (ifAssigned.length !== 0) {
+    throw Error(
+      `Course ${courseCode} has already been assigned with a curriculum!\n`
+    );
+  }
+  if (checkCourse.length === 0) {
```

Comment:
If you use my count suggestion above this would become 
```
```
```suggestion
  if (checkCourse == 0 ){
```

Reply:
follow up my reply above, `if (checkCourse === 0 )` won't work since it returns a data package

Reply:
Check my reply above on accessing the datapacket